packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -195,8 +204,19 @@ export const useUpdate = <RecordType extends RaRecord = any, ErrorType = Error>(
                     'useUpdate mutation requires a non-empty data object'
                 );
             }
+            if (getMutateMiddlewares) {
+                const updateMiddleware = getMutateMiddlewares(
```

Comment:
It seems we never need to pass around the `mutateWithMiddlewares` callback - we just need to call it. The API would be clearer like this:

```
if (hasMutationMiddlewares) {
   return mutateWithMiddlewares(dataProvider.update, resource, params);
} else {
  return dataProvider.update(resource, params);
}
```